Extensions/TileMap/tilemapruntimeobject.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * Displays a tilemap (mapeditor.org supported).
+ * @memberof gdjs
+ * @class TileMapRuntimeObject
+ * @extends RuntimeObject
+ */
+gdjs.TileMapRuntimeObject = function(runtimeScene, objectData) {
+  gdjs.RuntimeObject.call(this, runtimeScene, objectData);
+  this._frameElapsedTime = 0;
+
+  /** @type {number} */
+  this._opacity = objectData.content.opacity;
+  /** @type {string} */
+  this._tilemapJsonFile = objectData.content.tilemapJsonFile;
+  /** @type {string} */
+  this._tilemapAtlasImage = objectData.content.tilemapAtlasImage;
+  /** @type {string} */
+  this._displayMode = objectData.content.displayMode;
+  /** @type {number} */
+  this._layerIndex = objectData.content.layerIndex;
+  /** @type {number} */
+  this._animationSpeed = objectData.content.animationSpeed;
+  /** @type {number} */
+  this._animationFps = objectData.content.animationFps;
+
+  if (this._renderer)
+    gdjs.TileMapRuntimeObjectRenderer.call(this._renderer, this, runtimeScene);
+  else
+    this._renderer = new gdjs.TileMapRuntimeObjectRenderer(this, runtimeScene);
+
+  // *ALWAYS* call `this.onCreated()` at the very end of your object constructor.
+  this.onCreated();
+};
+
+gdjs.TileMapRuntimeObject.prototype = Object.create(
+  gdjs.RuntimeObject.prototype
+);
+gdjs.registerObject('TileMap::TileMap', gdjs.TileMapRuntimeObject);
+
+gdjs.TileMapRuntimeObject.prototype.getRendererObject = function() {
+  return this._renderer.getRendererObject();
+};
+
+gdjs.TileMapRuntimeObject.prototype.update = function(runtimeScene) {
+  if (this._animationSpeed <= 0) return;
+  var elapsedTime = this.getElapsedTime(runtimeScene) / 1000; 
+  
+  this._frameElapsedTime += elapsedTime * this._animationSpeed; 
+  // 0.25 = 4 fps , 0.0625 = 1  fps
+  var normalizedFps = this._animationFps * 0.0625;
```

Comment:
Why 0.0625? I think you did a wrong calculation here.

A "fps" is a frame *per* second. 
So you must take the inverse: `while (this._frameElapsedTime > 1/this._animationFps)`
See the code for Sprite objects.

Reply:
I just divided 0.25 by 4 , because you told me that 0.25 equals 4 fps. Where is the code for sprite objects? 
I found this in spriteruntimeobject.js
```
       var elapsedTime = this.getElapsedTime(runtimeScene) / 1000;
        this._frameElapsedTime += this._animationPaused ? 0 : elapsedTime * this._animationSpeedScale;

        if ( this._frameElapsedTime > direction.timeBetweenFrames ) {
            var count = Math.floor(this._frameElapsedTime / direction.timeBetweenFrames);
            this._currentFrame += count;
            this._frameElapsedTime = this._frameElapsedTime-count*direction.timeBetweenFrames;
            if ( this._frameElapsedTime < 0 ) this._frameElapsedTime = 0;
        }
```
Note how it doesnt have any fps in the equation. I am starting to think that maybe I should take out the fps from the extension and just have timeBetweenFrames?

Reply:
It's still not good. If you're putting a magic number like this, it's certain that there is something wrong :)

Either do:
- `while (this._frameElapsedTime > 1/this._animationFps) ` if you want to use fps.
- or `while (this._frameElapsedTime > this._animationTimeBetweenFrames`) if you want to use a time in seconds between each frame (like done in sprites).

Reply:
I did, can you have a look at 
https://github.com/4ian/GDevelop/pull/1901/files#diff-aa352dd5abdc5a41e91ff51a3d1a324e3212c78c36c256a9d6f1509a651a1309R44
I think you were viewing an old commit yesterday

Reply:
Weird, I guess I've seen the commit that was originally used for the comment. All good now! :)